packet.go

```diff
@@ -4,187 +4,171 @@ import (
 	"bytes"
 	"encoding/binary"
 	"fmt"
+	"strings"
 )
 
+type Packet interface {
+	Unpack(data []byte) error
+	Pack() []byte
+}
```

Comment:
I haven't dig deep yet, but I so far I love the idea of this interface! Is it a python thingy?

Reply:
it's not my code ;(

Reply:
Oh lol, someone suck as a code reviewer :P

Reply:
Ah, lol, I'm slowpoke :) Some of that is the code from, like, `v1`? I had some wired flashback in the form of blast from the distant past, phew.